src/app/components/practice/question-authoring-page/question-authoring.guard.ts

```diff
@@ -0,0 +1,22 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, ActivatedRouteSnapshot, RouterStateSnapshot, UrlTree } from '@angular/router';
+import { UserPermission } from '../../../models/permissions';
+import { AuthenticationService } from '../../../services/authentication.service';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class QuestionAuthoringGuard implements CanActivate {
+  constructor(
+    private authenticationService: AuthenticationService
+) { }
+  canActivate(
+    next: ActivatedRouteSnapshot,
+    state: RouterStateSnapshot): boolean {
+      let isPermitted  = false;
+      this.authenticationService.currentUser$.subscribe( x => x.permissions.
+           has(UserPermission.ORG_ADMIN) ? isPermitted = true : isPermitted = false);
+      return isPermitted;
```

Comment:
I'm not sure if this variable will be set by the time the synchronous execution has happened.

Reply:
It turns out, https://angular.io/api/router/CanActivate, you can return an observable with a Boolean in it and it will be happy :). 

This means you can use the `.pipe` operator instead of subscribe and return the result of the Boolean operation you're doing within the operator.